### PR TITLE
Update handling events 749

### PR DESCRIPTION
**Pull request by tsonny**

### CODE REVIEW

File: docs/src/modules/java/pages/eventsourced.adoc [hunk 1]
```diff
@@ -61,11 +61,11 @@ include::example$java-eventing-shopping-cart/src/main/java/shopping/cart/Shoppin
...
 
-IMPORTANT: The **only** way for a command handler to modify Entity state is by emitting an event. Any modifications made directly to the state from the command handler are not persisted. When the Entity is passivated and reloaded, those modifications will not be present.
+When the command handler receives a request, the event is emitted and Akka Serverless immediately sends the event to the journal, and invokes the entity event handler.  Immediately after, the entity state is updated to the new value and can be used to create the reply. 
 
-The following example of a command handler emits an event:
+NOTE: An entity update event can occur before the event has been stored in the journal. This does not cause any concurrency issue because, Akka Serverless only process one command at a time per entity instance. The next command is not processed until an event from the previous command has been successfully been stored.
```
Comment (pvlugter) [inline docs/src/modules/java/pages/eventsourced.adoc:68]: As mentioned in the akkaserverless-docs PR, the original text makes more sense to me in this part. Maybe just leave out these particular changes.

The Java SDK is being changed and will have a different approach for emitting events and using the updated state.

**Review by pvlugter — CHANGES_REQUESTED** [review]

**Review by octonato — APPROVED** [review]
Looks good to me, but I agree with Peter that the explanation about emitting events will change soon and that we will need to revisit it anyway.

File: docs/src/modules/java/pages/eventsourced.adoc (hunk 1 above)
Comment (tsonny) [reply docs/src/modules/java/pages/eventsourced.adoc]: fixed: 500af37472fd58d0f872f30dcc80a12faf285798

File: docs/src/modules/java/pages/eventsourced.adoc [hunk 2]
```diff
@@ -61,12 +61,10 @@ include::example$java-eventing-shopping-cart/src/main/java/shopping/cart/Shoppin
 
 == Emitting events
 
-Commands can modify state by emitting events. A command handler can emit an event by taking in a link:{attachmentsdir}/api/com/akkaserverless/javasdk/eventsourcedentity/CommandContext.html[`CommandContext` {tab-icon}, window="new"] parameter, and invoking the link:{attachmentsdir}/api/com/akkaserverless/javasdk/eventsourcedentity/CommandContext.html#emit(java.lang.Object)[`emit` {tab-icon}, window="new"] method on it. Invoking `emit` will immediately invoke the associated event handler for that event. This validates that the event can be applied to the current state, and also updates state for command handler processing.
+Commands can modify state by emitting events. A command handler can emit an event by taking in a link:{attachmentsdir}/api/com/akkaserverless/javasdk/eventsourcedentity/CommandContext.html[`CommandContext` {tab-icon}, window="new"] parameter, and invoking the link:{attachmentsdir}/api/com/akkaserverless/javasdk/eventsourcedentity/CommandContext.html#emit(java.lang.Object)[`emit` {tab-icon}, window="new"] method on it.
```
Comment (pvlugter) [inline docs/src/modules/java/pages/eventsourced.adoc:64]: May be useful to keep the information about event handlers being invoked immediately. Adding back this text, extended a little:

```suggestion
Commands can modify state by emitting events. A command handler can emit an event by taking in a link:{attachmentsdir}/api/com/akkaserverless/javasdk/eventsourcedentity/CommandContext.html[`CommandContext` {tab-icon}, window="new"] parameter, and invoking the link:{attachmentsdir}/api/com/akkaserverless/javasdk/eventsourcedentity/CommandContext.html#emit(java.lang.Object)[`emit` {tab-icon}, window="new"] method on it.

Invoking `emit` will immediately invoke the associated event handler for that event. This validates that the event can be applied to the current state, and also makes the updated state available for creating the response message or any other subsequent processing in the current command handler.

**Review by pvlugter — APPROVED** [review]

**Comment by pvlugter** [comment]
@tsonny, just a note to use the "Squash and merge" option when merging a PR like this (rather than the "Create a merge commit" option).

**Pull request merged**